packages/table/src/TableContext.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { createContext, useContext, useMemo } from 'react';
+
+// type Action
+const Types = {
+  SelectableTable: 'SELECTABLE_TABLE',
+  ToggleHeaderCheckedState: 'TOGGLE_HEADER_CHECKED',
+  ToggleHeaderIndeterminate: 'TOGGLE_HEADER_INDETERMINATE',
+  SetColumnInfo: 'SET_COLUMN_INFO',
+  SortTableData: 'SORT_TABLE_DATA',
+  ToggleIndividualChecked: 'TOGGLE_INDIVIDUAL_CHECKED',
+  SetRowCheckedState: 'SET_ROW_CHECKED_STATE',
+  SetHasNestedRows: 'SET_HAS_NESTED_ROWS',
+  SetHasRowSpan: 'SET_HAS_ROW_SPAN',
+} as const;
+
+type Types = typeof Types[keyof typeof Types];
+
+export { Types };
+
+interface ActionPayload {
+  [Types.SelectableTable]: boolean;
+  [Types.ToggleHeaderCheckedState]: boolean | undefined;
+  [Types.ToggleHeaderIndeterminate]: boolean;
+  [Types.SetColumnInfo]: {
+    dataType?: DataType;
+    index: number;
+  };
+  [Types.SetHasRowSpan]: boolean;
+  [Types.SetHasNestedRows]: boolean;
+  [Types.SortTableData]: {
+    columnId: number;
+    key: string;
+    data: Array<unknown>;
+  };
+  [Types.ToggleIndividualChecked]: {
+    index: number;
+    checked?: boolean;
+  };
+  [Types.SetRowCheckedState]: Array<boolean | undefined>;
+}
+
+type ActionMap<A extends Record<string, any>> = {
+  [Key in keyof A]: A[Key] extends undefined
+    ? {
+        type: Key;
+      }
+    : {
+        type: Key;
+        payload: A[Key];
+      };
+};
+
+type Action = ActionMap<ActionPayload>[keyof ActionMap<ActionPayload>];
+
+// type Dispatch
+type Dispatch = (action: Action) => void;
+
+// type State
```

Comment:
This comment is misplaced? Or you're breaking up sections of the file?

I'd suggest being bolder if that's the case:

```
//
// State
//

```

Reply:
Was breaking up sections, but I can eliminate if you don't think its necessary

Reply:
👍